plugins/outputs/elasticsearch/elasticsearch.go

```diff
@@ -220,9 +220,15 @@ func (a *Elasticsearch) Connect() error {
 		Timeout:   time.Duration(a.Timeout),
 	}
 
+	elasticURL, err := url.Parse(a.URLs[0])
+	if err != nil {
+		return err
+	}
```

Comment:
As far as I see this is the approach used in the "official", isn't it? However, I'm not so sure it's a good one. Imagine, by chance, the first URL has `http` but all other URLs do have `https`... This will then throw away the majority of your nodes. Wouldn't it be better to either

1. Do a majority vote on the scheme.
2. Group the URLs by scheme and use multiple clients.

or a combination thereof?

Reply:
@srebhan all elastic search nodes in a cluster are required to use the same connection, either TLS is enabled for the cluster, or it is disabled for the cluster. Clients using the official libraries from Elastic also only ever support either TLS is enabled or TLS is disabled. You can NOT mix and match http vs https in the same cluster.

Most people will use a singular URL when sniffing (which may be a load balancer that sends requests to any one node that is up) and then sniffing will find all the other nodes.

Reply:
@archoversight understood. Thanks for clarifying this!